### PR TITLE
fix: unnecesary double loops

### PR DESCRIPTION
**Pull request by Carlos-err406**
## Pull Request Checklist

- [x] **Description:** Briefly describe the changes in this pull request.
- [x] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [x] **Documentation:** Have you updated relevant documentation?
- [ ] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?

---

## Description
addresses issue #966 but for [this](https://github.com/open-webui/open-webui/issues/966#issuecomment-1971900475) comment
---

### Added

- [List any new features or additions]

### Fixed
unnecesary double loops in several parts of the codebase

### Changed
src/lib/components/chat/Settings/Audio.svelte
src/lib/components/ChangelogModal.svelte
src/lib/components/chat/Settings/Models.svelte
src/routes/(app)/prompts/+page.svelte

### Removed

- [List any removed features or files]

### CODE REVIEW

**Comment by Carlos-err406** [comment]
i think those are all the {#each x.filter() as y} loops (^^")

**Comment by Carlos-err406** [comment]
hey @tjbck could you check on this if you have the chance? thanks in advance

File: src/lib/components/layout/Sidebar.svelte [hunk 1]
```diff
@@ -346,195 +346,185 @@
...
-							>
-								<div class=" flex self-center flex-1 w-full">
-									<div
-										class=" text-left self-center overflow-hidden {chat.id === $chatId
-											? 'w-[160px]'
-											: 'w-full'} "
-									>
-										{chat.title}
-									</div>
+				{#each $chats as chat, i}
```
Comment (tjbck) [inline src/lib/components/layout/Sidebar.svelte:349]: Thanks for the PR, I'm not sure if this approach is more readable in this specific case.

**Comment by tjbck** [comment]
Thanks for taking a look. I do see the possible performance gain but I'm not entirely sure if this is necessary, as filtering and mapping combination is wildly used and I believe it's more readable with the current approach. I'll close this issue/PR for now and maybe we can come back to this later.

File: src/lib/components/layout/Sidebar.svelte (hunk 1 above)
Comment (Carlos-err406) [reply src/lib/components/layout/Sidebar.svelte:349]: should i revert the sidebar one?

**Pull request closed without merging**